pkg/launcher/uninstall/uninstall.go

```diff
@@ -0,0 +1,54 @@
+package uninstall
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"os"
+
+	"github.com/kolide/launcher/ee/agent"
+	"github.com/kolide/launcher/ee/agent/types"
+)
+
+func Uninstall(ctx context.Context, k types.Knapsack) {
+	uninstallNoExit(ctx, k)
+
+	if err := disableAutoStart(ctx); err != nil {
+		k.Slogger().Log(ctx, slog.LevelError,
+			"disabling auto start",
+			"err", err,
+		)
+	}
+
+	// TODO: remove start up files
+	// TODO: remove installation
+
+	os.Exit(0)
+}
+
+// uninstallNoExit just removes the enroll secret file and wipes the database without
+// exiting the process. This is so that we can test a portion of the uninstall process.
```

Comment:
```suggestion
// uninstallNoExit just removes the enroll secret file and wipes the database without
// exiting the process. This is so that we can test a portion of the uninstall process. This
// logs errors, but does not return them, because we want to try each uninstall independently
```

Reply:
I think?

Reply:
done, except replaced "try each uninstall" with "try each step"